manager/oracles.js

```diff
@@ -637,39 +637,40 @@ const instagram = async (UserId, link) => {
         var fbPage = await FbPage.findOne({
             instagram_username: instagramUserName,
         })
-
         if (fbPage && fbPage.instagram_id) {
             var instagram_id = fbPage.instagram_id
             var fbProfile = await FbProfile.findOne({ UserId: UserId }).lean()
             if (fbProfile) {
                 var accessToken = fbProfile.accessToken
                 var mediaGetNewAccessToken = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/oauth/access_token?grant_type=fb_exchange_token&client_id=${process.env.APPID}&client_secret=${process.env.APP_SECRET}&fb_exchange_token=${accessToken}`
-                var resMediaAccessToken = (await rp.get(mediaGetNewAccessToken)).data
-                var media =
-                    'https://graph.facebook.com/' +
-                    oauth.facebook.fbGraphVersion +
-                    '/' +
-                    instagram_id +
-                    '/media?fields=like_count,shortcode,media_url&limit=50&access_token=' +
-                    resMediaAccessToken.access_token
-                var resMedia = (await rp.get(media)).data
-                var data = resMedia.data
+                var resMediaAccessToken = (await rp.get(mediaGetNewAccessToken))
+                var media = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/${instagram_id}/media?fields=like_count,shortcode,media_url&limit=50&access_token=${resMediaAccessToken.data.access_token}`
+                var resMedia = (await rp.get(media))
+                var data = resMedia.data.data
```

Comment:
In the current code, there are several HTTP API calls all embedded directly in your main function. This makes the function lengthy and hard to follow, and it could also make reusing these API calls more difficult in the future. 

A better approach would be to move these API calls into their own functions. This way, each function would have a single responsibility (in line with the Single Responsibility Principle), making the code more readable, easier to maintain, and easier to test.

For instance, you might have a function for getting a new access token from Facebook. Here's what that could look like:

```javascript
async function getNewAccessToken(accessToken, fbGraphVersion) {
    const mediaGetNewAccessToken = `https://graph.facebook.com/${fbGraphVersion}/oauth/access_token?grant_type=fb_exchange_token&client_id=${process.env.APPID}&client_secret=${process.env.APP_SECRET}&fb_exchange_token=${accessToken}`;
    const resMediaAccessToken = await rp.get(mediaGetNewAccessToken);
    return resMediaAccessToken.data.access_token;
}
```

Then, you could use this function in your main function:

```javascript
if (fbProfile) {
    const accessToken = await getNewAccessToken(fbProfile.accessToken, oauth.facebook.fbGraphVersion);
    // ...
}
```

By following this approach for all of the API calls in your function, you can make your code much cleaner and more modular. Each individual function is easier to understand, and if you ever need to change how you get a new access token, for instance, you would only have to do it in one place in your code.